pkg/azureclient/client.go

```diff
@@ -113,6 +121,60 @@ func (c *azureClient) StartVirtualMachine(ctx context.Context, resourceGroup, na
 	return c.virtualMachinesClient.Start(ctx, resourceGroup, name)
 }
 
+// GetHyperVGenerationVersion gets the HyperVGeneration version for the given disk instance type. Defaults to V2 if either V1 or V2
+// available.
+func (c *azureClient) GetHyperVGenerationVersion(ctx context.Context, instanceType string, diskType string, region string) (version string, err error) {
```

Comment:
There probably is but I don't think we want to introduce using the installer copy in this PR because it sounds like something we'd want to do more broadly. Wdyt?

Reply:
Per Meet™: We would need to construct a client from github.com/openshift/installer/pkg/asset/installconfig/azure, which we don't do today. Queue it up for future tech debt reduction. Because copypasta sucks.

Reply:
HIVE-1937